azext_edge/edge/providers/orchestration/resources/connector/opcua/certs.py

```diff
@@ -0,0 +1,363 @@
+        }
+
+        spc_object = self._process_fortos_yaml(object_text=spc_object, secret_entry=secret_entry)
+
+        if not spc:
+            logger.warning(f"Azure Key Vault Secret Provider Class {OPCUA_SPC_NAME} not found, creating new one...")
+            spc_client_id = spc_properties.get("clientId", "")
+            spc_tenant_id = spc_properties.get("tenantId", "")
+            spc = {
+                "location": self.instance["location"],
+                "extendedLocation": self.instance["extendedLocation"],
+                "properties": {
+                    "clientId": spc_client_id,  # The client ID of the service principal
+                    "keyvaultName": spc_keyvault_name,
+                    "tenantId": spc_tenant_id,
+                    "objects": spc_object,
+                },
+            }
+        else:
+            spc["properties"]["objects"] = spc_object
+
+        with console.status(f"Updating secret reference in Secret Provider Class {OPCUA_SPC_NAME}..."):
+            poller = self.ssc_mgmt_client.azure_key_vault_secret_provider_classes.begin_create_or_update(
+                resource_group_name=resource_group,
+                azure_key_vault_secret_provider_class_name=OPCUA_SPC_NAME,
+                resource=spc,
+            )
+            wait_for_terminal_state(poller)
+
+    def _add_secret_to_secret_sync(
+        self,
+        secret_name: str,
+        file_name: str,
+        secret_sync: dict,
+        resource_group: str,
+        spc_name: str,
+        secret_sync_name: str,
+    ) -> dict:
+        # check if there is a secret sync called secret_sync_name, if not create one
+        secret_mapping = secret_sync.get("properties", {}).get("objectSecretMapping", [])
+        # add new secret to the list
+        secret_mapping.append(
+            {
+                "sourcePath": secret_name,
+                "targetKey": file_name,
+            }
+        )
+
+        # find duplicate targetKey
+        target_keys = [mapping["targetKey"] for mapping in secret_mapping]
+        if len(target_keys) != len(set(target_keys)):
+            logger.error("Cannot have duplicate targetKey in objectSecretMapping.")
+            return
+
+        if not secret_sync:
+            logger.warning(f"Secret Sync {secret_sync_name} not found, creating new one...")
+            secret_sync = {
+                "location": self.instance["location"],
+                "extendedLocation": self.instance["extendedLocation"],
+                "properties": {
+                    "kubernetesSecretType": "Opaque",
+                    "secretProviderClassName": spc_name,
+                    "serviceAccountName": SERVICE_ACCOUNT_NAME,
+                    "objectSecretMapping": secret_mapping,
+                },
+            }
+        else:
+            secret_sync["properties"]["objectSecretMapping"] = secret_mapping
+
+        # create a new secret sync
+        with console.status("Updating Secret Sync..."):
```

Comment:
Hi @c-ryan-k @vilit1 the ultimate goal is to set up some ux similar to init and create command for the multisteps, but due to the time for M3, now I just have multiple console.status to indicate which step we are at... any thoughts for this? thanks a lot!

Reply:
I think that's ok for now, let's get the initial prototype working and see how it looks, but I don't think it's an issue to start with a limited UI at first